### DISCUSSION TITLE
matcher multiple patterns

### DISCUSSION THREAD

**Discussion by tbright17** [explosion/spaCy#8092] [Help: Coding & Implementations] [answered]
I have multiple patterns for matching:
p1,p2,p3,p4
if I use 
```
matcher.add("some id",[p1,p2,p3,p4])
```
and I have a doc that matches both p1,p3,p4. But I find every time I call the matcher, it only return matches with p3.
I'm wondering if there is any order when multiple patterns are used. What if I want to return matches in order? Like in this case, I want matches with p1.

**Comment by polm** [comment, marked as answer]
If you are calling a matcher explicity it should return all matches. If you are using an EntityRuler it will give priority to longer mentions, and in the case of the same length, to earlier mentions. ([docs](https://spacy.io/usage/rule-based-matching#:~:text=if%20any%20matches%20were%20to%20overlap%2C%20the%20pattern%20matching%20most%20tokens%20takes%20priority.%20if%20they%20also%20happen%20to%20be%20equally%20long%2C%20then%20the%20match%20occurring%20first%20in%20the%20doc%20is%20chosen.)). 

Can you give full example code?
